p4-16/spec/P4-16-spec.mdk

```diff
@@ -3710,7 +3710,9 @@ only implicitly casts from `int` to fixed-width types and from enums
 with an underlying type to the underlying type. In particular,
 applying a binary operation to an expression of type `int` and an
 expression with a fixed-width type will implicitly cast the `int`
-expression to the type of the other expression.
+expression to the type of the other expression. For enums with an underlying
+type, it can be implicitly cast to its underlying type whenever appropriate,
+including but not limited to in shift, concatenation, bit slicing, header stack indexing and other binary operations.
```

Comment:
Rewrite to remove implication that casts are inserted when a serializable enum is used as the base of a bit slice here.